aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncMockingTest.java

```diff
@@ -0,0 +1,376 @@
+     * the mock will immediately callback on the provided onStart callback.
+     */
+    @Test
+    public void onStartCallbackIsCalledForMockOnCreate() {
+        AppSyncMocking.onCreate(appSync).callOnStart();
+        Completable
+            .create(subscriber -> appSync.onCreate(
+                Model.class,
+                subscriptionToken -> subscriber.onComplete(),
+                NoOpConsumer.create(),
+                NoOpConsumer.create(),
+                EmptyAction.create()
+            ))
+            .test()
+            .awaitDone(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+            .assertComplete();
+    }
+
+    /**
+     * callOnStart() arranges a mock behavior on the provided AppSync instance.
+     * When the mock is called upon to create a subscription for update events,
+     * the mock will immediately callback on the provided onStart callback.
+     */
+    @Test
+    public void onStartCallbackIsCalledForMockOnUpdate() {
+        AppSyncMocking.onUpdate(appSync).callOnStart();
+        Completable
+            .create(subscriber -> appSync.onUpdate(
+                Model.class,
+                subscriptionToken -> subscriber.onComplete(),
+                NoOpConsumer.create(),
+                NoOpConsumer.create(),
+                EmptyAction.create()
+            ))
+            .test()
+            .awaitDone(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+            .assertComplete();
+    }
+
+    /**
+     * callOnStart() arranges a mock behavior on the provided AppSync instance.
+     * When the mock is called upon to create a subscription for delete events,
+     * the mock will immediately callback on the provided onStart callback.
+     */
+    @Test
+    public void onStartCallbackIsCalledForMockOnDelete() {
+        AppSyncMocking.onDelete(appSync).callOnStart();
+        Completable
+            .create(subscriber -> appSync.onDelete(
+                Model.class,
+                subscriptionToken -> subscriber.onComplete(),
+                NoOpConsumer.create(),
+                NoOpConsumer.create(),
+                EmptyAction.create()
+            ))
+            .test()
+            .awaitDone(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+            .assertComplete();
+    }
+
+    private static Predicate<GraphQLResponse<ModelWithMetadata<BlogOwner>>> equalsRelaxingTimestamp(
+            GraphQLResponse<ModelWithMetadata<BlogOwner>> expectedResponse) {
+        return response -> {
+            if (!ObjectsCompat.equals(expectedResponse.getErrors(), response.getErrors())) {
+                return false;
+            }
+            ModelWithMetadata<BlogOwner> expectedMwm = expectedResponse.getData();
+            ModelWithMetadata<BlogOwner> mwm = response.getData();
+            return equalsRelaxingTimestamp(expectedMwm).test(mwm);
+        };
+    }
+
+    private static Predicate<ModelWithMetadata<BlogOwner>> equalsRelaxingTimestamp(
```

Comment:
Ick... so what's the issue exactly for needing this workaround?

Reply:
In `AppSyncMocking`, I have a couple of flavors of the `mockSuccessResponse(...)` which internally create their own `Temporal.Timestamp.now()`. (So, I can't know what it is, outside, in this unit test.)

However, I just realized that I could test the other flavors of `mockSuccessResponse(...)`, instead -- the ones that _will_ allow me to supply a known timestamp. That would get rid of these kludges and clean things up quite a bit. I'll do that.